### PR TITLE
Jupyter Notebook to Blog Post GitHub Action

### PR DESCRIPTION
**Pull request by hamelsmu**
First of all, thank you to @sgugger for putting up with several questions and making amendments to the nbdev library to make this possible.  

This GitHub Action uses nbdev to convert notebooks to HTML, and saves these as .md files in `_posts` so that they can be rendered by GitHub pages.  The motivation is that folks can write their blog posts as notebooks and simply save these notebooks into a folder and it magically becomes a blog post.  

### High-Level Overview
1.  Save a Jupyter Notebook into the `_notebooks` folder, preferably prepended with `YYYY-MM-DD` as you would do with a markdown post.  
2.  The authors should put the following front matter in the notebook per usual 
```
# Title
> Summary
- metadata_key1: metadata_value1
- metadata_key2: metadata_value2 
```

3.  **If you fail to use the right naming convention**, GitHub Actions will automatically prepend `YYYY-MM-DD` to the `.md` file placed in `_posts`, which is inferred by the file's last modified date.  Even though this file has a `.md` extension, it is really a HTML file with front matter (depending on your point of view).
4.  This workflow triggers only when a change to a file in `_notebooks` is detected on the master branch.
5. **UPDATE**:  It could be annoying to have Actions automatically do a notebook conversion on every push to master since that would cause your local environment to have conflicts and you would have to pull after every push when editing notebooks.  In order to circumvent this problem, **this workflow only triggers when the magic word `/sync` appears anywhere in your commit messages** ( see [filter](https://github.com/fastai/fast_template/pull/5/files#diff-fd48c7c54988c6b2d00efd3eaf20d1abR11) ).  Also, as mentioned above this workflow will only trigger when you push to the master branch.  We can change and customize any of these behaviors as you wish.
6. Since the conversion is done as HTML, interactive charts and graphs in Altair are preserved.  (Still trying to figure out ipywidgets)

### Other Notes / Discussion:
- Right now if the destination file is prepended with `YYYY-MM-DD` automatically because the notebook is not named correctly, the only feedback that the user gets are the logs in Actions.  Perhaps I should have the Action open a GitHub issue informing the user that they haven't followed convention and the date was inferred.  ( Note that the notebook's names are never changed, only the converted file).
- when an error occurs with notebook2html, it only prints the error to the screen, however it might be nice to raise the error so that the workflow fails and an alert is thrown via the normal mechanisms? I made a PR to address in case it is helpful https://github.com/fastai/nbdev/pull/65

### CODE REVIEW

File: .github/workflows/nb2post.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    steps:
+    - name: Copy Repository Contents
+      uses: actions/checkout@master
+
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.6'
+        architecture: 'x64'
+    
+      # These are the minimal steps I could come up with install nbdev in the actions runner
```
Comment (hamelsmu) [inline .github/workflows/nb2post.yaml:22]: @sgugger perhaps you would be interested in my install steps, I tried to make it as minimal as possible but these are the steps I found to make it work for me.  It still installs very very fast, though so no worries.

File: .github/workflows/nb2post.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - '_notebooks/*.ipynb'
+
+jobs:
+  convert-notebook:
+    if: contains(join(github.event.commits.*.message), '/sync')
```
Comment (hamelsmu) [inline .github/workflows/nb2post.yaml:11]: FYI this is one of the coolest parts about Actions which allow you to introspect the metadata of the events that trigger workflows.  In this line, we are checking to see if any of the commit messages in the push that triggers the workflow contain the string `/sync`

**Comment by jph00** [comment]
This is looking great @hamelsmu ! :) My only suggestion is to remove the sample notebook. Instead, put a README in the _notebooks folder telling people what that folder is there for, and how to use it.

Most folks using fast_template won't be blogging with notebooks, and won't know what they are, so I don't want to have them see a sample notebook post appear in their initial blog - I think that would be a bit intimidating!

**Comment by hamelsmu** [comment]
Sounds good, still working on it will let you know when it's ready for review (I'll finish it by end of today)

**Comment by hamelsmu** [comment]
@jph00 I made the following changes you suggested:

- Added `_notebooks/README.md`
- Removed `_notebooks/*ipynb` and any correlated markdown files from `_posts/`
- Removed installation of PIL from setup steps

Please let me know if there is anything else I can add/change that would be useful!  Thank you 🙇

**Comment by jph00** [comment]
Fantastic!

**Pull request merged**